### PR TITLE
resolve Issue 82: Improve error for watermark error

### PR DESCRIPTION
**Pull request by technomad21c**
added vault name, operation type, level, and chain ID to error logs

- test
```
$ curl -XPOST      -d '"029caecab9db439a2dd462570a929bef822cb94441b80f9f728faeb1b3d9235250770f1e8f0000061ddc"'      localhost:6732/keys/tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg

{"signature":"sigPpWB6xgH37RiZQ64dpXtAotDP6EQ2uuarPeq8aw6LRVe3WAVES3LQV6midPcNBN1RT3JA4ZCfBkFgrP7TKeRaMJEbE71m"}

$ curl -XPOST      -d '"029caecab9db439a2dd462570a929bef822cb94441b80f9f728faeb1b3d9235250770f1e8f0000061ddc"'      localhost:6732/keys/tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg

{"error":"Not safe to sign"}
```
```
$ curl -XPOST      -d '"019caecab900061de402e27da655a04eaa5dad0647e6ff56d11a5da8efb48c2e90570e27853839e76b68000000005eb576f5047ab08836902391c075dc92640f9d7496faa8cff5b2b24450786d86349b9a528d000000110000000101000000080000000000061de3ad348c90c42bc5b90e89837fdbeb6c1360be7181c9116ef2eb8cb63ebbb1380e00000675d7e0ffa2030000"'      localhost:6732/keys/tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg

{"signature":"sigtGikfyVYaRuRwEaXG6q464zS8N6G3tAngJBrESoUzJ8ZrFJmDMACEPTvGtTLTgL43bh9qCi2BKW7rzrie82hTjNWfKCkS"}

$ curl -XPOST      -d '"019caecab900061de402e27da655a04eaa5dad0647e6ff56d11a5da8efb48c2e90570e27853839e76b68000000005eb576f5047ab08836902391c075dc92640f9d7496faa8cff5b2b24450786d86349b9a528d000000110000000101000000080000000000061de3ad348c90c42bc5b90e89837fdbeb6c1360be7181c9116ef2eb8cb63ebbb1380e00000675d7e0ffa2030000"'      localhost:6732/keys/tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg

{"error":"Not safe to sign"}
```

- results
```
ERRO[0004] Error signing request: Not safe to sign       chain_id=NetXjD3HPJJjmcd lvl=400860 op=endorsement pkh=tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg vault_name=File
```
```
ERRO[1594] Error signing request: Not safe to sign       chain_id=NetXjD3HPJJjmcd lvl=400868 op=block pkh=tz1SVwdLNf3ANMQE1AXrxS1pBG8tcn2joVZg vault_name=File
```

### CODE REVIEW

File: secret.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ { "name": "brandons_wallet",
```
Comment (jevonearth) [inline secret.json:1]: This should not be added to git.

File: signatory.yaml [hunk 2]
```diff
@@ -6,35 +6,15 @@ server:
 
 vaults:
   # Name is used to identify backend during import process
-  kms:
-    driver: cloudkms
+  local_file_keys:
```
Comment (jevonearth) [inline signatory.yaml:9]: Please restore the original example configuration file.

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (jevonearth) [inline test.sh:1]: This testing file is helpful, but it lacks context. Remove this for now, and we will make real integration tests.

File: pkg/server/server.go [hunk 4]
```diff
@@ -60,7 +68,23 @@ func (s *Server) signHandler(w http.ResponseWriter, r *http.Request) {
 
 	signature, err := s.Signer.Sign(r.Context(), keyHash, data)
 	if err != nil {
-		s.logger().Errorf("Error signing request: %v", err)
+		p, _ := s.Signer.GetPublicKey(r.Context(), keyHash)
```
[the remark this answers is not in the document]
Comment (technomad21c) [reply pkg/server/server.go:71]: Agreed. I added the error handling for GetPublicKey.

File: pkg/server/server.go [hunk 5]
```diff
@@ -60,7 +68,23 @@ func (s *Server) signHandler(w http.ResponseWriter, r *http.Request) {
...
+		vaultName := p.VaultName
+		msg, _ := tezos.ParseUnsignedMessage(data)
```
[the remark this answers is not in the document]
Comment (technomad21c) [reply pkg/server/server.go:73]: added the error handling for ParseUnsignedMessage.

File: secret.json (hunk 1 above)
Comment (technomad21c) [reply secret.json:1]: Sorry, I didn't recognize this was added to git commit.  This was removed.

File: signatory.yaml (hunk 2 above)
Comment (technomad21c) [reply signatory.yaml:9]: I restored the original one.

File: test.sh (hunk 3 above)
Comment (technomad21c) [reply test.sh:1]: Agreed. The file was removed.

**Pull request merged**